test.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const rehype = require('rehype');
+const dedent = require('dedent');
+const rehypeHighlightCodeBlock = require('.');
+
+const processHtml = html => {
+  return rehype()
+    .data('settings', { fragment: true })
+    .use(rehypeHighlightCodeBlock, {
+      highlight: jest.fn(() => '<span>mockSyntaxHighlighting</span>')
+    })
+    .processSync(html)
+    .toString();
+};
+
+describe('rehypeHighlightCodeBlock', () => {
+  test('mocks syntax highlighting properly', () => {
+    const result = processHtml(dedent`
+      <div>
+        <p>foo</p>
+        <pre><code>p { color: red }</code></pre>
+        <p>bar</p>
+      </div>
+    `);
+    expect(result).toMatchSnapshot();
```

Comment:
In addition to these snapshot checks, we should also check and make sure that the spy function (created with `jest.fn()`) was called with the correct arguments.